### ISSUE TITLE
Not saving my API key for danbooru.

### ISSUE THREAD

**Issue by execb5** [ahodesuka/ahoviewer#20]
I don't know if the API key was supposed to be saved in the preferences, but I have to enter it everytime I launch ahoviewer. The username is saved though...

I only use Danbooru, don't know if this happens with other boorus.

Also, this gets printed out when the application is launched:
```
** Message: Remote error from secret service: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.secrets was not provided by any .service files
Failed to lookup password for Danbooru
  The name org.freedesktop.secrets was not provided by any .service files
```

If saving the API key wasn't supposed to be a feature, then I'm sorry for opening this issue.

**Comment by ahodesuka** [comment]
If you want passwords/API keys to be stored securely you need to install `gnome-keyring`.
It will provide the missing .service file.

If you do not care about storing it securely you can configure ahoviewer with `--disable-libsecret`.

**Comment by execb5** [comment]
I installed `gnome-keyring` but it still doesn't save the key.

When I put the key in the preferences window I get the output:

```
** Message: Remote error from secret service: org.freedesktop.DBus.Error.UnknownMethod: No such interface 'org.freedesktop.Secret.Collection' on object at path /org/freedesktop/secrets/collection/login
Failed to set password for Danbooru
  No such interface 'org.freedesktop.Secret.Collection' on object at path /org/freedesktop/secrets/collection/login
```

**Comment by ahodesuka** [comment]
What distro are you using?
Your package manager is probably at fault, and not pulling in all the dependencies needed.

Maybe try restarting gnome-keyring with
```
pkill gnome-keyring-daemon
gnome-keyring-daemon -s -f -c secrets
```

**Comment by execb5** [comment]
I'm using Arch.

I've managed to set up the keyring and everything is working now!
The problem was that I never actually had the daemon running, since I'm not using any DE or login manager.
I just had to follow the [console login](https://wiki.archlinux.org/index.php/GNOME_Keyring#Console_login) steps on the wiki (in case anyone ends up here with the same problem).

Thanks for your help and sorry for the inconvenience.
